### PR TITLE
RECIPE-42

### PR DESCRIPTION
**Pull request by PatsiChang**
Added token requirement to register new recipe (Including recipe Icon upload)
Link: https://smart-home-cloud.atlassian.net/browse/RECIPE-42

### CODE REVIEW

File: src/main/java/com/patsi/controller/RecipeController.java [hunk 1]
```diff
@@ -32,12 +34,31 @@ public class RecipeController {
     private RecipeService recipeService;
     @Autowired
     private RecipeRegistrationValidator recipeRegistrationValidator;
+    @Autowired
+    private UserProfileService userProfileService;
+
+    @GetMapping("getRecipeByToken")
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/controller/RecipeController.java]: I suggest to name it as something like `getMyRecipe`

File: src/main/java/com/patsi/service/RecipeService.java [hunk 2]
```diff
@@ -29,37 +29,37 @@ public class RecipeService {
     @Value("${spring.web.resources.static-locations[3]}")
     private String IMAGE_PATH;
 
+    //Get Existing Recipe
+    public List<Recipe> getRecipe(String uid) {
+        List<Recipe> finalRecipeList = recipeRepository.findAll().stream()
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/service/RecipeService.java]: avoid findAll. Should only query recipe by uid.

Image there are million of recipe, the memory will explode

File: src/main/java/com/patsi/service/RecipeService.java [hunk 3]
```diff
@@ -29,37 +29,37 @@ public class RecipeService {
...
+        log.info("In Service: updateRecipeIcon");
+        FileHelper.newFile(IMAGE_PATH, id, recipeIcon);
     }
 
-    public void updateRecipeIcon(UUID id, byte[] recipeIcon) throws IOException {
-        FileHelper.newFile(IMAGE_PATH, id.toString(), recipeIcon);
-        Recipe recipe = recipeRepository.findById(id).get();
-        recipe.setImgURL(id.toString());
+    //Update Existing Recipe
+    public void updateRecipe(UUID id, Recipe recipe) {
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/service/RecipeService.java:56]: Need validation on the recipe,  e.g. owner of the recipe, any mandatory field is missing, etc...

Can raise a ticket and implement it after refinement

Comment (PatsiChang) [reply src/main/java/com/patsi/service/RecipeService.java:56]: created ticket for this 
Link: https://smart-home-cloud.atlassian.net/browse/RECIPE-50

File: src/main/java/com/patsi/service/RecipeService.java (hunk 2 above)
Comment (PatsiChang) [reply src/main/java/com/patsi/service/RecipeService.java]: Fixed

File: src/main/java/com/patsi/controller/RecipeController.java (hunk 1 above)
Comment (PatsiChang) [reply src/main/java/com/patsi/controller/RecipeController.java]: Fixed

**Review by ChakShingLo — APPROVED** [review]
lgtm

**Pull request merged**